### DISCUSSION TITLE
DigiByte Core v8.22.0-RC2

### DISCUSSION THREAD

**Discussion by JaredTate** [DigiByte-Core/digibyte#127] [Announcements]
# DigiByte Blockchain v8.22.0- rc2 Release Notes

We are excited to announce the test release of DigiByte Blockchain v8.22.0 rc1. This release is based on Bitcoin Core v22 and brings in various features and improvements from Bitcoin Core v0.18, v0.19, v0.20, v0.21, and v22. Please note that this is a test release and not intended for production use.

## Testing and Feedback

As this is a test release, we encourage users to try out the new features and improvements and provide feedback on any issues you encounter. This will help us refine the final release for a more stable and feature-rich DigiByte Blockchain experience.  

Please report any [issues](https://github.com/DigiByte-Core/digibyte/issues) or [feedback](https://github.com/DigiByte-Core/digibyte/discussions) on our GitHub repository or join the DigiByte community on Gitter, Telegram, Discord, or Reddit.

We would like to thank our developers and the entire DigiByte community for their continued support and dedication to the project. Together, we are making the DigiByte Blockchain more secure, efficient, and accessible for everyone.

## Notable Changes

- **Minimum Transaction Fee Increase**: The minimum transaction fee has been increased from 0.001 DGB to 0.1 DGB per kilobyte.

- **Taproot Deployment Delay**: Please note that the Taproot deployment date has been set to 2099 for this release to delay activation. Given the immense amount of changes in this release, it was decided by core developers to delay taproot activation until the next DGB core release to make sure there were no inadvertent vulnerabilities introduced allowing the DGB network to be easily spammed by taproot changes.

- **Bitcoin Core Feature Inclusions**: Major changes from Bitcoin Core v0.18, v0.19, v0.20, v0.21, and v22 have been integrated into DigiByte v8.22.0 rc1. Some highlights include:
- 
  - **Taproot and Schnorr Signatures (BTC v0.21)**: This release includes the implementation of Taproot and Schnorr signatures, which bring various improvements to the DigiByte Blockchain, such as increased privacy, enhanced smart contract capabilities, and more efficient signature aggregation. Although the functionality is fully integrated, the activation dates for Taproot and Schnorr signatures have been delayed until 2099. The activation dates will be updated in the next release to enable these features for the DigiByte community.

  - **Partial Spend Support (BTC v0.18)**: This feature allows users to partially spend their wallet's UTXO (Unspent Transaction Output) set, improving privacy and efficiency.
  
  - **Block Filter Indexing (BTC v0.19)**: This enables light clients to more efficiently and privately query the blockchain for their relevant transactions.
  
  - **BIP 158 Compact Block Filters (BTC v0.19)**: This feature allows clients to synchronize more efficiently while maintaining a high degree of privacy.
  
  - **BIP 339 WTxID Relay (BTC v0.21)**: WTxID-based transaction announcements reduce the bandwidth used during transaction relay and improve transaction propagation.
  
  - **Descriptor Wallets (BTC v0.21)**: A new wallet format using output script descriptors, which simplifies wallet management and backups.
  
  - **Hardware Wallet Interface (HWI) Integration (BTC v0.18)**: Improved hardware wallet support, making it easier for users to securely store and manage their DigiByte holdings.
  
  - **Anchor Connections (BTC v0.20)**: This feature allows nodes to maintain persistent connections to specific peers, improving network stability and resistance to eclipse attacks.
  
  - **Multi-Wallet Support (BTC v0.18)**: This feature enables users to manage multiple wallets simultaneously within the DigiByte Core GUI. Users can now load, create, and unload wallets on demand without having to restart the software.
  
  - **Major RPC Changes**: Numerous RPC changes and improvements have been implemented across various Bitcoin Core versions, including new commands, optimizations, and enhanced error handling. Some notable RPC command changes are:
  
      - `getblocktemplate` (BTC v0.18): Added support for the `coinbasetxn` field.

      - `getblockreward` (DGB v8.22): Returns the most recent mining block reward for DGB.
      
      - `createwallet` (BTC v0.18): Introduced to create new wallets dynamically.
      
      - `unloadwallet` (BTC v0.18): Introduced to unload wallets without restarting the software.
      
      - `getbalances` (BTC v0.19): A new command that returns an object with all balances.
      
      - `getdescriptorinfo` (BTC v0.21): Provides information about a given descriptor.
      
      - `joinpsbts` (BTC v0.20): Combines multiple PSBTs (Partially Signed Bitcoin Transactions) into one.
  
- **Codebase Improvements**: Over 285,000 lines of code have been changed in this release, making the DigiByte Blockchain more efficient, secure, and robust.

## Who should upgrade?

Anyone wishing to help us fully test v8.22.0 rc1.

## What's Changed
* Update: GPG key for ycagel by @ChillingSilence in https://github.com/DigiByte-Core/digibyte/pull/49
* Update Qt 5.9 source URL by @Jolly-Pirate in https://github.com/DigiByte-Core/digibyte/pull/50
* Docs: Replaced IRC references with Gitter references by @gto90 in https://github.com/DigiByte-Core/digibyte/pull/54
* small wording changes by @mctrivia in https://github.com/DigiByte-Core/digibyte/pull/59
* WIP: Feature/8.22.0 by @gto90 in https://github.com/DigiByte-Core/digibyte/pull/62
* Add receive socket fds by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/64
* Refactor DandelionShuffleThread to run in connection scheduler by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/65
* DGB Seed Nodes Update  by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/66
* Basic deployment maintenance by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/68
* Add new DNS seed - seed.digibyte.help by @saltedlolly in https://github.com/DigiByte-Core/digibyte/pull/67
* Dandelion transactions by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/69
* fix some urls in depends by @digicontributer in https://github.com/DigiByte-Core/digibyte/pull/70
* URL fix: https://dgbwiki.com/ by @GreenParticle in https://github.com/DigiByte-Core/digibyte/pull/74
* Fix "Unknown new rules activated" Error & Uninitialized Variable nMinerConfirmationWindow, Revert Changes by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/76
* Prevent algo bits from triggering 'unknown new rules' message by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/75
* Dandelion removal by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/79
* Reimplement Dandelion Transactions by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/80
* Disable RBF transaction signalling by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/82
* Remove legacy odocrypt fork code by @barrystyle in https://github.com/DigiByte-Core/digibyte/pull/86
* Fix: Dandelion mutex LOCK usages by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/84
* Core Improvements and Bugfixes by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/89
* Fix: Functional Tests by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/90
* Update Unit Test Vectors by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/93
* WIP: Fix Functional Tests by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/94
* Fixed remaining Functional tests for 8.22 by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/106
* Remove non-functioning seeders and full nodes by @saltedlolly in https://github.com/DigiByte-Core/digibyte/pull/104
* WIP: Feature/8.22.0 by @gto90 in https://github.com/DigiByte-Core/digibyte/pull/63
* Reward to 0 Code by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/105
* New mainnet & testnet seed server. by @Jongjan88 in https://github.com/DigiByte-Core/digibyte/pull/107
* Add-Seed-Node by @Quakeitup in https://github.com/DigiByte-Core/digibyte/pull/109
* This PR introduces support for GUIX cross compilations by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/108
* Update chainparams.cpp by @bastiandriessen in https://github.com/DigiByte-Core/digibyte/pull/110
* Update DGB SVG Image by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/112
* Add Jared Tate GPG Key by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/113
* GUI fixes by replacing icons and improving the splash screen by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/114
* Disable taproot until we find a solution to prevent chain bloats by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/115
* Fix tests related to subsidy by @SmartArray in https://github.com/DigiByte-Core/digibyte/pull/116
* Update Copyright Year & Wallet Image by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/117
* Merge Develop to RC1 by @JaredTate in https://github.com/DigiByte-Core/digibyte/pull/118

## New Contributors
* @Jolly-Pirate made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/50
* @mctrivia made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/59
* @JaredTate made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/66
* @saltedlolly made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/67
* @GreenParticle made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/74
* @Jongjan88 made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/107
* @Quakeitup made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/109
* @bastiandriessen made their first contribution in https://github.com/DigiByte-Core/digibyte/pull/110

**Full Changelog**: https://github.com/DigiByte-Core/digibyte/compare/v7.17.3...v8.22.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/DigiByte-Core/digibyte/releases/tag/v8.22.0-rc2'>DigiByte Core v8.22.0-RC2</a>.</em>

**Comment by ZULUPooL** [comment]
![Screenshot from 2023-07-27 23-01-46](https://github.com/DigiByte-Core/digibyte/assets/140152241/d16f1732-90bd-4f69-b36d-590ab12de3fa)
All algorithms are smoothly accepting blocks in the testnet, yet I wouldn't call the test fully complete. To comprehensively assess its performance, we should introduce more spam transactions. Our v8.22.0 rc1 testnet mining is still ongoing [here](https://dgb-t.zulupool.com/)  and if you can facilitate spam transactions, it would be immensely helpful in evaluating its functionality.

For those in need of testnet coins, feel free to share your address, and we'll gladly send them your way; we've already accumulated a considerable amount.

Additionally, if anyone finds the odocrypt CPU miner useful, you can access it [here](https://github.com/ZULUPooL/cpuminer-opt)

**Reply by saltedlolly** [reply]
Thank you for your work on this. If you have a surplus of DGBT coins, it might be helpful for the community to send them to Renzo's Testnet Faucet: https://digifaucet.org/testnet

With a larger supply he can increase the payout per transaction. This will make it easy for anyone who needs DGBT to get some without having to mine on testnet themselves.

**Reply by ZULUPooL** [reply]
I can't send the coins.
The first time I did it - not all transactions went away and we forked the testnet. 
We rolled back the nodes, and try to send coins from another wallet, the transaction appears in the log, but it does not appear in the block explorer and we get the fork again... 
I will sync testnet from scratch just in case, and try again when it's ready.

**Reply by ZULUPooL** [reply]
Well, it's a puzzling situation. 
Whenever we mine in the testnet and attempt to send coins, it results in a testnet fork. 
On the other hand, if we refrain from mining in the testnet and solely try to send coins, the transaction never seems to appear in the testnet block explorer. 
Our suspicion is that this occurs when the transaction size is significantly large, nearly reaching the maximum number of inputs.

As an interesting experiment, if anyone wishes to explore the limits of the testnet, we've provided the private key of the wallet used for attempting to send 230k coins. `eg8D9yfYWn6xxZfMLR4dhBprmJLxtYXD1zEcUNHRc65GyHkxDCQY`

Here is our config - maybe something is wrong there, but even if it is, it shouldn't break the blockchain.

```
blockreconstructionextratxn=1000
dbcache=1024

chain=test
fallbackfee=0.002
algo=skein
maxsendbuffer=5000
daemon=1
listen=1
server=1
#reindex=1
[test]
rpcbind=127.0.0.1
rpcbind=192.168.100.105
rpcallowip=127.0.0.1
rpcallowip=192.168.100.1/24
port=33351
rpcport=33350
rpcworkqueue=1000
```

`DigiByte Core version v8.22.0-gaf42429717acd26d64afaf26fa26e58b7ff6ad6a (release build)`

**Reply by ZULUPooL** [reply]
Without making any changes, the system has begun sending transactions of any size when our mining is turned off. On the other hand, if we continue mining all 5 algorithms simultaneously, we still encounter testnet forks. It's possible that there is no specific error, and the behavior might be linked to the fact that we are mining all 5 algorithms concurrently. Given the rapid block mining with 5 running nodes on a single server, it's worth noting that the nodes have only a limited number of external peers, typically around 3-4. 
Any thoughts or ideas on this matter?

**Reply by gto90** [reply]
I think you are correct @ZULUPooL, that when you start mining all five algos you fork the testnet chain because you are outperforming any other testnet mining.

@JaredTate , @SmartArray , or @digicontributer do you have additional thoughts or ideas here?
